### ISSUE TITLE
A non-trivial issue in update ApplicationContext

### ISSUE THREAD

**Issue by msylvester** [uchicago-mobi/2016-Spring-Forum#96]
I want to use updateApplicaitonContext as the watch is only receiving the most up to date info and the past does not matter.  With that said the code below does not work as intended.  Note the dictionary parameter ([String:AnyObject]) is a misnomer, as far as I can tell.  

**session.updateApplicationContext([String:AnyObject])**

[String:String] 👌🏽
[String:Int] 👌🏽
[String:UIIMage] 👎
[String:NSData]  👎

I have even tried serializing the data without success.  Has anyone been able to use this effectively?

Note also, shared App groups between iPhone and appleWatch seems to be deprecated.  https://forums.developer.apple.com/thread/3927

**Comment by tabinks** [comment]
You can only use Plist compatible types, which explains the UIImage.  NSData should work.

That thread is an eye opener for me.  Although, it makes sense.  You don't need to store the data yourself if you are using Watch connectivity. When you update the application context the system automatically handle storing that information until it's ready to send the watch. There is nothing that the watch would have to read data from. All the data should be stored in the application context.

**Comment by msylvester** [comment]
Well, I have had success using Update Application Context passing a UIImage as NSData or simply a UIIMage serialized, BUT ONLY when the image is local in the assets folder. 

HOWEVER,  when pulling an image from cloud kit, I cannot send it to the watch.  This code fails.  Any permutation of getting a CKAsset from CloudKit and sending to the watch fails.   The code that fails, looks like:

if let img = self.records[indexPath.row].valueForKey("imageView") {
   let  image = NSData(contentsOfFile: img.fileURL.path!)!
}
WatchSessionManager.sharedManager.updateApplicationContext(["image" : image])

I have used reflection to verify the type of 'image' above.  I have compressed the Image without success as well. Has anyone pulled a CKAsset from cloud kit and sent it to the watch with success?
